docs/source/user-guide/explain-usage.md

```diff
@@ -0,0 +1,644 @@
+```
+
+Figure 3: A typical query plan of leading edge (most recent) data
+
+Let us begin reading bottom up. The bottom or leaf nodes are always either `ParquetExec` or `RecordBatchExec`.
+There is a `ParquetExec` leaf in this plan. Let's go over it.
+
+### Bottom leaf operator: `ParquetExec`
+
+**First ParqetExec**
+
+```sql
+|               |                       ParquetExec: file_groups={10 groups: [[/data/hits.parquet:0..1477997645], [/data/hits.parquet:1477997645..2955995290], [/data/hits.parquet:2955995290..4433992935], [/data/hits.parquet:4433992935..5911990580], [/data/hits.parquet:5911990580..7389988225], ...]}, projection=[EventTime, RegionID, OS], predicate=to_timestamp(EventTime@4) >= 200000000000 AND to_timestamp(EventTime@4) < 700000000000 AND RegionID@8 = 839, pruning_predicate=CASE WHEN RegionID_null_count@2 = RegionID_row_count@3 THEN false ELSE RegionID_min@0 <= 839 AND 839 <= RegionID_max@1 END, required_guarantees=[RegionID in (839)] |
+```
+
+Figure 4: `ParquetExec` leaf node
+
+* This `ParquetExec` includes 10 groups. Each group can contain one or many files but, in this example, there
+  is
+  one file for many groups. Files in each group are read sequentially but groups will be executed in parallel. So for
+  this
+  example, 10 groups will be read in parallel.
+* `/data/hits.parquet`: this is the path of the file.
+* `projection=[EventTime, RegionID, OS]`: there are many columns in this table but only these 4 columns are read.
+*
+
+`predicate=to_timestamp(EventTime@4) >= 200000000000 AND to_timestamp(EventTime@4) < 700000000000 AND RegionID@8 = 839`:
+filter in the query used for data pruning,
+
+*
+
+`pruning_predicate=CASE WHEN RegionID_null_count@2 = RegionID_row_count@3 THEN false ELSE RegionID_min@0 <= 839 AND 839 <= RegionID_max@1 END`:
+the actual
+pruning predicate transformed from the predicate above. This is used to filter files outside that predicate.
+
+* `required_guarantees=[RegionID in (839)]`: TODO, need definition.
+
+### Data-scanning structures
+
+So the question is why we split parquet files into different ParquetExec while they are in the same partition? There are
+many reasons but two major ones are:
+
+1. Split the non-overlaps from the overlaps so we only need to apply the expensive deduplication operation on the
+   overlaps. This is the case of this example.
+2. Split the non-overlaps to increase parallel execution
+
+##### When we know there are ovelaps?
```

Comment:
@alamb can overlaps happen in normal datafusion query plans? Or is this specific to InfluxDB? If possible could you point me in the direction of getting a parquet data-set with overlaps so I can run the EXPLAIN's locally?

Reply:
Actually taking a look at the datafusion docs I'm not seeing `DeduplicateExec` in the public crate docs so I'm assuming this is not valid for core datafusion. 
https://docs.rs/datafusion/latest/datafusion/physical_plan/union/struct.UnionExec.html?search=DeduplicateExec